### PR TITLE
gnuplot: make readline optional

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/gnuplot/package.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ class Gnuplot(AutotoolsPackage):
         description="Enable PBM (Portable Bit Map) and other older bitmap terminals",
     )
     variant("qt", default=False, description="Build with QT")
+    variant("readline", default=True, description="Build with readline")
 
     # required dependencies
     depends_on("readline")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: Should this only depend on `readline` when `+readline`?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/gnuplot/package.py]: See https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-dependencies.

Comment (iarspider) [reply var/spack/repos/builtin/packages/gnuplot/package.py]: Thanks for the catch!

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**